app/models/user.rb

```diff
@@ -0,0 +1,19 @@
+class User < ActiveRecord::Base
+  include BCrypt
+
+  def self.register(username:, full_name:, email:, password:)
+    encrypted_password = Password.create(password)
+    user = self.create(username: username, full_name: full_name, email: email, password: encrypted_password)
+  end
+
+  def self.authenticate(username:, password:)
```

Comment:
Can you use `p` to get visibility on what is happening when a user tries to log in with the correct password and then checking the server logs.

For example, you might consider printing out:
- username
- password
- user
- user.password


With these printed out, can you see why the login feature isn't working?

Reply:
Yes!! Thank you I've got it.

I should have spotted this but was so confused by my positive tests. Once I solved the issue I figured out that I was accidentally creating a user with an unhashed password in one of my support files which was giving a false positive